src/commons/util/file.rs

```diff
@@ -46,81 +55,95 @@ pub fn file_path(base_path: &Path, file_name: &str) -> PathBuf {
+            io::Error::new(io::ErrorKind::Other, "could not deserialize json"),
+        )
+    })
 }
 
 /// Saves a file, creating parent dirs as needed
-pub fn save_in_dir(content: &Bytes, base_path: &Path, name: &str) -> Result<(), io::Error> {
+pub fn save_in_dir(content: &Bytes, base_path: &Path, name: &str) -> Result<(), KrillIoError> {
     let mut full_path = base_path.to_path_buf();
     full_path.push(name);
     save(content, &full_path)
 }
 
 /// Saves a file under a base directory, using the rsync uri to create
 /// sub-directories preserving the rsync authority and module in dir names.
-pub fn save_with_rsync_uri(content: &Bytes, base_path: &Path, uri: &uri::Rsync) -> Result<(), io::Error> {
+pub fn save_with_rsync_uri(content: &Bytes, base_path: &Path, uri: &uri::Rsync) -> Result<(), KrillIoError> {
     let path = path_with_rsync(base_path, uri);
     save(content, &path)
 }
 
 /// Reads a file to Bytes
-pub fn read(path: &Path) -> Result<Bytes, io::Error> {
-    let mut f = File::open(path).map_err(|_| Error::cannot_read(path))?;
+pub fn read(path: &Path) -> Result<Bytes, KrillIoError> {
+    let mut f =
+        File::open(path).map_err(|e| KrillIoError::new(format!("Could not read: '{}'", path.to_string_lossy()), e))?;
     let mut bytes = Vec::new();
-    f.read_to_end(&mut bytes)?;
+    f.read_to_end(&mut bytes)
+        .map_err(|e| KrillIoError::new(format!("Could not read: {}", path.to_string_lossy()), e))?;
     Ok(Bytes::from(bytes))
 }
 
-pub fn read_with_rsync_uri(base_path: &Path, uri: &uri::Rsync) -> Result<Bytes, io::Error> {
+pub fn read_with_rsync_uri(base_path: &Path, uri: &uri::Rsync) -> Result<Bytes, KrillIoError> {
     let path = path_with_rsync(base_path, uri);
     read(&path)
 }
 
-pub fn delete_with_rsync_uri(base_path: &Path, uri: &uri::Rsync) -> Result<(), io::Error> {
+pub fn delete_with_rsync_uri(base_path: &Path, uri: &uri::Rsync) -> Result<(), KrillIoError> {
     delete(&path_with_rsync(base_path, uri))
 }
 
-pub fn delete_in_dir(base_path: &Path, name: &str) -> Result<(), io::Error> {
+pub fn delete_in_dir(base_path: &Path, name: &str) -> Result<(), KrillIoError> {
     let mut full_path = base_path.to_path_buf();
     full_path.push(name);
     delete(&full_path)
 }
 
-pub fn delete(full_path: &Path) -> Result<(), io::Error> {
+pub fn delete(full_path: &Path) -> Result<(), KrillIoError> {
     trace!("Removing file: {}", full_path.to_string_lossy());
-    fs::remove_file(full_path)?;
-    Ok(())
+    fs::remove_file(full_path)
+        .map_err(|e| KrillIoError::new(format!("Could not remove file: {}", full_path.to_string_lossy()), e))
 }
 
-pub fn clean_file_and_path(path: &Path) -> Result<(), io::Error> {
+pub fn clean_file_and_path(path: &Path) -> Result<(), KrillIoError> {
     if path.exists() {
-        fs::remove_file(&path)?;
+        fs::remove_file(&path)
+            .map_err(|e| KrillIoError::new(format!("Could not remove file: {}", path.to_string_lossy()), e))?;
```

Comment:
The error message here is the same as in `fn delete()` above, will it be possible to distinguish these two cases from each other in the logs?

Reply:
done, and renamed `delete` to `delete_file` to be explicit.